search/engine.go

```diff
@@ -454,25 +509,52 @@ func joinResponses(sortBy string, from int, size int, results ...*elastic.Search
 	if len(results) == 0 {
 		return nil, nil
 	}
+
 	// Concatenate all result hits to single slice.
 	concatenated := make([]*elastic.SearchHit, 0)
 	for _, result := range results {
 		concatenated = append(concatenated, result.Hits.Hits...)
 	}
 
+	// Keep only unique results by MDB_UID (additional results with a duplicate MDB_UID might be added by Grammar). We keep the result with a higher score.
```

Comment:
This should be refactored to a function and tested as a black box.

Reply:
Nice

Reply:
done